src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -590,23 +590,27 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref HashS
         /// traversal stops and that type is returned from this method. Otherwise if traversal
         /// completes without the predicate returning true for any type, this method returns null.
         /// </summary>
+        /// <param name="useDefaultType">If true, use <see cref="TypeWithAnnotations.DefaultType"/>
+        /// instead of <see cref="TypeWithAnnotations.Type"/> to avoid early resolution of nullable types</param>
         public static TypeSymbol VisitType<T>(
             this TypeWithAnnotations typeWithAnnotationsOpt,
             TypeSymbol typeOpt,
             Func<TypeWithAnnotations, T, bool, bool> typeWithAnnotationsPredicateOpt,
             Func<TypeSymbol, T, bool, bool> typePredicateOpt,
             T arg,
-            bool canDigThroughNullable = false)
+            bool canDigThroughNullable = false,
+            bool useDefaultType = false)
         {
             Debug.Assert(typeWithAnnotationsOpt.HasType == (typeOpt is null));
+            Debug.Assert(canDigThroughNullable = false || useDefaultType == false, "digging through nullable will cause early resolution of nullable types");
```

Comment:
>canDigThroughNullable = false [](start = 25, length = 29)

I think this assignment is not intentional and is the source of the test failures.